packages/internal-testing-tools/src/report/Report.ts

```diff
@@ -0,0 +1,53 @@
+import type {
+    Assertion,
+    AssertionResult,
+    EventTypeAssertions
+} from '../types/report';
+
+export class Report {
+    assertions: AssertionResult[] = [];
+    status: 'FAILED' | 'PASSED' | 'NOT_RUN' = 'NOT_RUN';
+
+    // TODO: This typing enables to keep the arguments for each key.
+    // However the return type of assertions is void instead of boolean.
+    // This can be done by returning tuples for assertions instead of object.
+    connectAssertionsToReport = <AssertionsType extends EventTypeAssertions>(
+        assertions: AssertionsType
+    ): AssertionsType => {
```

Comment:
I believe this type is good enough for now. 
With this type we can get autocomplete on the assertions, but we're missing that it's returning a `boolean` (`true` if test succeeded, `false` otherwise). 

Since this is an internal package, I don't think this is a big deal